pkg/controllers/consolidation/controller.go

```diff
@@ -468,198 +525,130 @@ func (c *Controller) calculateLifetimeRemaining(node candidateNode) float64 {
 	return remaining
 }
 
-// nolint:gocyclo
-func (c *Controller) nodeConsolidationOptionReplaceOrDelete(ctx context.Context, node candidateNode) (consolidationAction, error) {
+//nolint:gocyclo
+func (c *Controller) nodeConsolidationOptionReplaceOrDelete(ctx context.Context, node candidateNode) (lifecyleCommand, error) {
```

Comment:
nit: It might be more apt to name this `computeNodeConsolidationOption` since it returns more than just a replace or delete action. Happy to do this in my follow up refactor though.